.github/workflows/test.yml

```diff
@@ -71,9 +73,12 @@ jobs:
         VTR_CMAKE_PARAMS: ${{ matrix.cmake }}
         NUM_CORES: ${{ matrix.cores }}
 
-    - uses: actions/upload-artifact@v3
-      if: ${{ always() }}
```

Comment:
I think this line should not have been removed.

Reply:
@AlexandreSinger : what do you think? Should this go back in?

Reply:
@soheilshahrouz  @vaughnbetz My bad... I misunderstood what this was trying to do. Perhaps in the future we should consider a better keyword to use:
https://docs.github.com/en/actions/learn-github-actions/expressions#always

In the meantime I have raised a PR reverting this removal: PR #2583